File: app/src/main/java/com/eagskunst/emmanuel/gamingnews/models/Cover.java
package com.eagskunst.emmanuel.gamingnews.models;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.annotations.SerializedName;

import java.lang.reflect.Type;

public class Cover{

	@SerializedName("id")
	private int id;

	@SerializedName("url")
	private String url;

	public Cover(){

	}

	public Cover(int id, String url) {
		this.id = id;
		this.url = url;
	}

	public Cover(int id) {
		this.id = id;
		this.url = null;
	}


	public void setId(int id){
		this.id = id;
	}

	public int getId(){
		return id;
	}

	public void setUrl(String url){
		this.url = url;
	}

	public String getUrl(){
		return url;
	}

	@Override
 	public String toString(){
		return 
			"Cover{" + 
			"id = '" + id + '\'' + 
			",url = '" + url + '\'' + 
			"}";
		}
}

File: app/src/main/java/com/eagskunst/emmanuel/gamingnews/models/Cover.kt
package com.eagskunst.emmanuel.gamingnews.models

import com.google.gson.annotations.SerializedName

class Cover() {

    @SerializedName("id")
    var id: Int = 0

    @SerializedName("url")
    var url: String? = null

    constructor(id: Int, url: String?) : this() {
        this.id = id
        this.url = url
    }

    constructor(id: Int) : this() {
        this.id = id
        this.url = null
    }

    override fun toString(): String {
        return "Cover{" +
                "id = '" + id + '\'' +
                ",url = '" + url + '\'' +
                "}"
    }
}
